pkg/devfile/parser/sourceAttribute.go

```diff
@@ -37,50 +39,74 @@ func AddSourceAttributesForTemplateSpecContent(sourceImportReference v1.ImportRe
+		parentOverrides.Projects[idx].Attributes.PutString(ParentOverrideAttribute, resolveImportReference(sourceImportReference))
 	}
-	for idx, project := range parentoverride.StarterProjects {
+	for idx, project := range parentOverrides.StarterProjects {
 		if project.Attributes == nil {
-			parentoverride.StarterProjects[idx].Attributes = attributes.Attributes{}
+			parentOverrides.StarterProjects[idx].Attributes = attributes.Attributes{}
 		}
-		parentoverride.StarterProjects[idx].Attributes.PutString(ImportSourceAttribute, fmt.Sprintf("parentOverrides from: %s", resolveImportReference(sourceImportReference)))
+		parentOverrides.StarterProjects[idx].Attributes.PutString(ParentOverrideAttribute, resolveImportReference(sourceImportReference))
 	}
 
 }
 
-// AddSourceAttributesForPluginOverride adds an attribute 'library.devfile.io/imported-from=<source reference>'
+// addSourceAttributesForPluginOverride adds an attribute 'library.devfile.io/imported-from=<source reference>'
 //  to all elements of plugin override that support attributes.
-func AddSourceAttributesForPluginOverride(sourceImportReference v1.ImportReference, pluginName string, pluginoverride *v1.PluginOverrides) {
-	for idx, component := range pluginoverride.Components {
+func addSourceAttributesForPluginOverride(sourceImportReference v1.ImportReference, pluginOverrides *v1.PluginOverrides) {
+	for idx, component := range pluginOverrides.Components {
 		if component.Attributes == nil {
-			pluginoverride.Components[idx].Attributes = attributes.Attributes{}
+			pluginOverrides.Components[idx].Attributes = attributes.Attributes{}
 		}
-		pluginoverride.Components[idx].Attributes.PutString(ImportSourceAttribute, fmt.Sprintf("pluginOverrides from: %s, plugin : %s", resolveImportReference(sourceImportReference), pluginName))
+		pluginOverrides.Components[idx].Attributes.PutString(PluginOverrideAttribute, resolveImportReference(sourceImportReference))
 	}
-	for idx, command := range pluginoverride.Commands {
+	for idx, command := range pluginOverrides.Commands {
 		if command.Attributes == nil {
-			pluginoverride.Commands[idx].Attributes = attributes.Attributes{}
+			pluginOverrides.Commands[idx].Attributes = attributes.Attributes{}
 		}
-		pluginoverride.Commands[idx].Attributes.PutString(ImportSourceAttribute, fmt.Sprintf("pluginOverrides from: %s, plugin : %s", resolveImportReference(sourceImportReference), pluginName))
+		pluginOverrides.Commands[idx].Attributes.PutString(PluginOverrideAttribute, resolveImportReference(sourceImportReference))
 	}
 
 }
+
+// AddSourceAttributesForOverrideAndMerge adds an attribute record the import reference to all elements of template that support attributes.
+func AddSourceAttributesForOverrideAndMerge(sourceImportReference v1.ImportReference, template interface{}) error {
+	if template == nil {
+		fmt.Errorf("cannot add source attributes to nil")
```

Comment:
return

now i want us to test this use case as well 🙃  do you think you can fit it in your current test cases, if you dont want a new test for it

Reply:
it is not going to fit in current test, as the template is never going to be `nil` when we call it from `parseParentAndPlugin`.

Reply:
updated